doc/connecting.md

```diff
@@ -7,12 +7,16 @@ or a pair of half-duplex `Stream` objects (e.g. STDIN and STDOUT streams). All A
 this document will assume a single bidirectional `Stream` for brevity.
 
 A JSON-RPC connection is created and managed via the `JsonRpc` class.
+
+## Connecting
+
 Certain decisions about the protocol details must be made up front while constructing the `JsonRpc` class.
-These decisions are documented in our [extensibility](extensibility.md) document.
-For this document, we will use the defaults for these up-front decisions.
-Other configurations can be changed after instantiating the `JsonRpc` class. We will review these in this document.
+This library includes several built-in protocol variants and options, and you can add your own. This is all documented in our [extensibility](extensibility.md) document.
 
-## Client
+In the remaining samples in this section we use the convenient static `Attach` method which
+instantiates the class with the default settings and begins listening for messages immediately. Samples for changing aspects of the protocol are further below in the next section.
```

Comment:
You may want to mention, or link to, the actual section. That way, if you ever insert another section in between, you don't have to remember to update this text.

Reply:
I wish I knew a good way to link _within_ a markdown document. I've seen ways documented, but haven't found one that works on github hosted docs.

Reply:
OK, I found at least one way. I've applied it.